libp2p/protocols/kademlia/xordistance.nim

```diff
@@ -31,25 +49,23 @@ proc `<`*(a, b: XorDistance): bool =
 proc `<=`*(a, b: XorDistance): bool =
   cmp(a, b) <= 0
 
-proc hashFor(k: Key): seq[byte] =
+proc hashFor(k: Key, hasher: Opt[XorDHasher]): seq[byte] =
```

Comment:
opt was chosen instead of default values due to my observation that this approach alerts the developer to unintentional malformed function calls. Requiring an explicit choice forces the developer to make a valid decision. Default values leaves the developer vulnerable to overlooking something that requires their attention